enumerable_methods.rb

```diff
@@ -0,0 +1,224 @@
+# rubocop:disable all 
+# 2
+module Enumerable
+  # 3
+  def my_each
+    if block_given?
+      iterator = 0
+      arr = self.to_a
+      while iterator < arr.length
+        yield(arr[iterator])
+        iterator += 1
+      end
+      return arr
```

Comment:
#my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self

In the case mentioned above, you have to return the range itself. Instead, you are returning an array.

Reply:
Hello @smcommits , thank you for the provided details.

Can you please send me the link of where is mentioned that (returning a range if it's passed)

As I can't see that here https://github.com/TheOdinProject/curriculum/blob/master/archive/old_lessons/ruby/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods

Nor in the enum method details here : https://ruby-doc.org/core-3.0.0/Enumerable.html

Reply:
@Ahmed-Benj This is actually the default behavior of each method in Ruby. The `each` method always returns the object it was invoked upon. In IRB, try calling each method on a range and pass a block. You will see that it will return the original range it was invoked upon. 

Please don't hesitate to ask if you have any other doubts. Good Luck!